lighthouse-core/gather/gatherers/forms.js

```diff
@@ -0,0 +1,185 @@
+    if (childElement instanceof HTMLInputElement || childElement instanceof HTMLTextAreaElement ) {
+      const inputAttributes = {
+        id: childElement.id,
+        nodeName: childElement.nodeName,
+        name: childElement.name,
+        placeholder: childElement.placeholder,
+        autocomplete: childElement.autocomplete,
+      };
+      formless.inputs.push(inputAttributes);
+    } else if (childElement instanceof HTMLSelectElement ) {
+      const selectAttributes = {
+        id: childElement.id,
+        nodeName: childElement.nodeName,
+        name: childElement.name,
+        placeholder: childElement.getAttribute('placeholder'),
+        autocomplete: childElement.autocomplete,
+      };
+      formless.inputs.push(selectAttributes);
+    } else if (childElement instanceof HTMLLabelElement) {
+      const labelAttributes = {
+        id: childElement.id,
+        nodeName: childElement.nodeName,
+        name: childElement.getAttribute('name'),
+        for: childElement.htmlFor,
+      };
+      formless.labels.push(labelAttributes);
+    }
+  }
+
+  return formless;
+}
+
+
+/**
+ * @return {LH.Artifacts['Forms']}
+ */
+/* istanbul ignore next */
+function collectFormElements() {
+  // @ts-ignore - put into scope via stringification
+  const formElements = getElementsInDocument('form'); // eslint-disable-line no-undef
+  // @ts-ignore - put into scope via stringification
+  const inputs = getElementsInDocument('input'); // eslint-disable-line no-undef
+  // @ts-ignore - put into scope via stringification
+  const labels = getElementsInDocument('labels'); // eslint-disable-line no-undef
+  // @ts-ignore - put into scope via stringification
+  const textareas = getElementsInDocument('textarea'); // eslint-disable-line no-undef
+  // @ts-ignore - put into scope via stringification
+  const selects = getElementsInDocument('select'); // eslint-disable-line no-undef
+
+  const formChildren = inputs.concat(labels, textareas, selects);
+  const formless = getFormlessElements(formChildren);
+
+  const forms = formElements.map(/** @param {HTMLFormElement} formElement */ (formElement) => {
+    const attributes = {
+      id: formElement.id,
+      name: formElement.name,
+      autocomplete: formElement.autocomplete,
+    };
+
+    return {
+      attributes: attributes,
+      inputs: getChildrenInputs(formElement),
+      labels: getChildrenLabels(formElement),
+    };
+  });
+
+  if (formless.inputs.length > 0 || formless.labels.length > 0) {
+    forms.push(formless);
+  }
+  return forms;
+}
+
+class Forms extends Gatherer {
+  /**
+   * @param {LH.Gatherer.PassContext} passContext
+   * @return {Promise<LH.Artifacts['Forms']>}
+   * @override
+   */
+  async afterPass(passContext) {
+    const driver = passContext.driver;
+
+    const expression = `(() => {
+      ${getChildrenInputs.toString()};
+      ${getChildrenLabels.toString()};
+      ${getFormlessElements.toString()};
+      ${pageFunctions.getOuterHTMLSnippetString};
```

Comment:
I think you can remove `getOuterHTMLSnippetString`, `isPositionFixedString`, and `getNodePathString` here since they don't seem to be used at all.